prov/verbs/src/verbs_info.c

```diff
@@ -502,8 +506,29 @@ int fi_ibv_fi_to_rai(const struct fi_info *fi, uint64_t flags,
 	rai->ai_qp_type = IBV_QPT_RC;
 	rai->ai_port_space = RDMA_PS_TCP;
 
-	if (!fi)
+	/* if no hints exists then use find_sysaddr info
+	 * as address provider */
+	if (!fi) {
+		struct fi_ibv_rdm_sysaddr iface_addr;
+		struct fi_ibv_rdm_sysaddr lo_addr;
+		fi_ibv_rdm_find_sysaddrs(&iface_addr, &lo_addr);
+
+                struct sockaddr_in *addr =
+		       	iface_addr.is_found ? &iface_addr.addr :
+			lo_addr.is_found ? &lo_addr.addr : NULL;
+
+		if(!addr)
+			return 0;
+
+		rai->ai_family = addr->sin_family;
+		rai->ai_flags |= RAI_FAMILY;
+
+		rai->ai_src_len = sizeof(*addr);
+		if (!(rai->ai_src_addr = malloc(rai->ai_src_len)))
+			return -FI_ENOMEM;
+		memcpy(rai->ai_src_addr, addr, rai->ai_src_len);
 		return 0;
+	}
```

Comment:
it is black magic in verbs initialization :)

I added fi_dup on every interface detected on system.